stock_portfolio/stock_portfolio/views/portfolio.py

```diff
@@ -62,12 +79,16 @@ def get_detail_view(request):
     try:
         symbol = request.matchdict['symbol']
     except KeyError:
-        return HTTPNotFound()
+        raise HTTPNotFound()
 
     try:
         query = request.dbsession.query(Stock)
-        stock_detail = query.filter(Stock.symbol == symbol).first()
+        stock_detail = query.filter(Junction.account_id == request.authenticated_userid).filter(Stock.symbol == symbol).one_or_none()
```

Comment:
Just as you have done above, this filter can be done in one `filter` by separating the conditions with commas.